Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs

```diff
@@ -150,47 +158,92 @@ private void ComputeControlGroupingIfNecessary()
-                        if (!partBinding.isPartOfComposite || partBinding.compositeOrCompositeBindingIndex != compositeBindingIndex)
-                            break;
-                        ++complexity;
-                    }
+	                var compositeBinding = bindingStates[binding.compositeOrCompositeBindingIndex];
+	                var composite = composites[compositeBinding.compositeOrCompositeBindingIndex];
+	                complexity = composite.GetPriority();
                 }
-                controlGroupingAndComplexity[i * 2 + 1] = (ushort)complexity;
+                controlGroupingAndComplexity[i].complexity = (ushort)complexity;
 
                 // Compute grouping. If already set, skip.
-                if (controlGroupingAndComplexity[i * 2] == 0)
+                if (controlGroupingAndComplexity[i].group == 0)
                 {
-                    if (!disableControlGrouping)
+                    for (var n = 0; n < totalControlCount; ++n)
                     {
-                        for (var n = 0; n < totalControlCount; ++n)
-                        {
-                            // NOTE: We could compute group numbers based on device index + control offsets
-                            //       and thus make them work globally in a stable way. But we'd need a mechanism
-                            //       to then determine ordering of actions globally such that it is clear which
-                            //       action gets a first shot at an input.
+                        // NOTE: We could compute group numbers based on device index + control offsets
+                        //       and thus make them work globally in a stable way. But we'd need a mechanism
+                        //       to then determine ordering of actions globally such that it is clear which
+                        //       action gets a first shot at an input.
 
-                            var otherControl = controls[n];
-                            if (control != otherControl)
-                                continue;
+                        var otherControl = controls[n];
+                        if (control != otherControl)
+                            continue;
 
-                            controlGroupingAndComplexity[n * 2] = (ushort)currentGroup;
-                        }
+                        controlGroupingAndComplexity[n].group = (ushort)currentGroup;
                     }
 
-                    controlGroupingAndComplexity[i * 2] = (ushort)currentGroup;
+                    controlGroupingAndComplexity[i].group = (ushort)currentGroup;
 
                     ++currentGroup;
                 }
+
+				// add the action that the current binding belongs to to the relevant group, sorted by complexity
+                if (controlGroupingAndComplexity[i].group >= m_ActionGroups.Length)
+                    ArrayHelpers.Resize(ref m_ActionGroups, Mathf.Max(10, m_ActionGroups.Length + 10), Allocator.Persistent);
```

Comment:
What's the reason for the resizing to the minimum of 10? Could we make this value a const so it's easier to refer to?

Reply:
From our sync, I understood that's a "random value" to avoid resizing multiple times with only 1 array. Still, making it a const makes sense to me, but it's not a blocking issue for this PR.